app/forms/schools/add_participant_form.rb

```diff
@@ -6,20 +6,52 @@ class AddParticipantForm
     include ActiveModel::Attributes
     include ActiveModel::Serialization
 
+    STEPS = %i[type details not_implemented].freeze
+
     TYPE_OPTIONS = {
       ect: "Early Career Teacher",
       mentor: "Mentor",
     }.freeze
 
+    attribute :completed_steps, default: []
     attribute :type
+    attribute :full_name
+    attribute :email
 
     validates :type,
               on: :type,
               presence: { message: "Please select type of the new participant" },
               inclusion: { in: TYPE_OPTIONS.keys.map(&:to_s), allow_blank: true }
 
+    validates :full_name,
+              on: :details,
+              presence: true
+
+    validates :email,
+              on: :details,
+              presence: true,
+              format: { with: Devise.email_regexp, allow_blank: true }
+
     def type_options
       TYPE_OPTIONS
     end
+
+    def previous_step(current_step)
+      return completed_steps.last unless current_step.in?(completed_steps)
+
+      completed_steps[completed_steps.index(current_step) - 1]
+    end
```

Comment:
There's a few issues with this which I suspect would be uncovered by splitting it out so other forms can reuse it and properly testing it

The main ones I've noticed is that on the first page it just goes to the page you're already on, and if you press back twice you end up in a two page loop